Description: Add keywords key to .desktop files
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Forwarded: https://github.com/mate-desktop/mate-user-share/issues/19
--- a/data/mate-user-share-properties.desktop.in
+++ b/data/mate-user-share-properties.desktop.in
@@ -7,6 +7,7 @@
 Terminal=false
 Type=Application
 Categories=Settings;X-MATE-NetworkSettings;
+Keywords=MATE;personal;file;sharing;preferences;properties;settings;WebDAV;ObexFTP;
 OnlyShowIn=MATE;
 X-MATE-Bugzilla-Bugzilla=MATE
 X-MATE-Bugzilla-Product=mate-user-share
